### PR TITLE
Add SST packages and dependencies

### PR DESCRIPTION
**Pull request by jjwilke**
* Remove deprecated SST packages
* Add dependents for SST elements

### CODE REVIEW

File: var/spack/repos/builtin/packages/dramsim2/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from spack import *
+
+
+class Dramsim2(MakefilePackage):
+    """
+    DRAMsim is a hardware-validated, cycle-accurate
+    C based simulator for DRAM devices such as DDR3
+    """
+
+    git = "https://github.com/umd-memsys/DRAMSim2"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dramsim2/package.py:16]: Needs a homepage

File: var/spack/repos/builtin/packages/dramsim2/package.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    version('2.2',
+            url="https://github.com/dramninjasUMD/DRAMSim2/archive/v2.2.2.tar.gz",
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dramsim2/package.py]: The URL says 2.2.2, but the version says 2.2. What's up with that? Also, can you move the URL up above like the git?

File: var/spack/repos/builtin/packages/dramsim2/package.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            sha256="96d0257eafb41e38ffa4f13e3ef3759567bdde7fa3329403f324abd0ddf8d015")
+
+    def install(self, spec, prefix):
+        install_tree("", prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dramsim2/package.py]:
```suggestion
        install_tree(".", prefix)
```

File: var/spack/repos/builtin/packages/dramsim2/package.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    def build(self, spec, prefix):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dramsim2/package.py:23]: I would move build before install

**Review by adamjstewart — CHANGES_REQUESTED** [review]
It would be much easier to review this PR if every package was in a separate PR. The comments I made to the first package apply to most of the packages, so I'll wait until they've all been updated before reviewing again.

File: var/spack/repos/builtin/packages/dramsim2/package.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    """
+    DRAMsim is a hardware-validated, cycle-accurate
+    C based simulator for DRAM devices such as DDR3
+    """
+
+    homepage = "https://github.com/umd-memsys/DRAMSim2"
+    git = "https://github.com/umd-memsys/DRAMSim2"
+
+    version('2.2.2',
+            url="https://github.com/dramninjasUMD/DRAMSim2/archive/v2.2.2.tar.gz",
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dramsim2/package.py]: Can you move this URL up below the `git` above?

File: var/spack/repos/builtin/packages/goblin-hmc-sim/package.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    """
+
+    homepage = "https://github.com/tactcomplabs/gc64-hmcsim"
+    git = "https://github.com/tactcomplabs/gc64-hmcsim"
+
+    version('1.0', url="https://github.com/tactcomplabs/gc64-hmcsim/archive/sst-8.0.0-release.tar.gz",
+            sha256="8a5e6b701865a581f15965d3ddd8c7d301b15f4b63543c444058e9c3688fd2c8")
+
+    def build(self, spec, prefix):
+        make()
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/goblin-hmc-sim/package.py]: You don't need to override `build` if you're just going to use the default

File: var/spack/repos/builtin/packages/hybridsim/package.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    depends_on("dramsim2")
+    depends_on("nvdimmsim")
+    patch("makefile.patch", when="@2.0.1")
+
+    def install(self, spec, prefix):
+        install_tree(".", prefix)
+
+    def build(self, spec, prefix):
+        symlink(spec["dramsim2"].prefix, "DRAMSim2")
+        symlink(spec["nvdimmsim"].prefix, "NVDIMMSim")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hybridsim/package.py:30]: Why is this necessary?

File: var/spack/repos/builtin/packages/intel-pin/package.py [hunk 8]
```diff
@@ -19,6 +19,10 @@ class IntelPin(Package):
     version('3.11', sha256='aa5abca475a6e106a75e6ed4ba518fb75a57549a59f00681e6bd6e3f221bd23a', url='https://software.intel.com/sites/landingpage/pintool/downloads/pin-3.11-97998-g7ecce2dac-gcc-linux.tar.gz')
     version('3.10', sha256='7c8f14c3a0654bab662b58aba460403138fa44517bd40052501e8e0075b2702a', url='https://software.intel.com/sites/landingpage/pintool/downloads/pin-3.10-97971-gc5e41af74-gcc-linux.tar.gz')
     version('3.7',  sha256='4730328795be61f1addb0e505a3792a4b4ca80b1b9405acf217beec6b5b90fb8', url='https://software.intel.com/sites/landingpage/pintool/downloads/pin-3.7-97619-g0d0c92f4f-gcc-linux.tar.gz')
+    version('2.14',
+            url="https://software.intel.com/sites/landingpage/pintool/downloads/pin-2.14-71313-gcc.4.4.7-linux.tar.gz",
+            sha256="1c29f589515772411a699a82fc4a3156cad95863a29741dfa6522865d4d281a1"
+            )
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/intel-pin/package.py]: You can keep this all on a single line like the other versions

File: var/spack/repos/builtin/packages/nvdimmsim/package.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    homepage = "https://github.com/slunk/NVDIMMSim"
+    git = "https://github.com/slunk/NVDIMMSim"
+
+    version('2.0.0', url="https://github.com/jimstevens2001/NVDIMMSim/archive/v2.0.0.tar.gz",
+            sha256="2a621ef10be5e52a1f543985d08354a2e6ee6532b5720e5f17ad6362cfd4adef")
+
+    def install(self, spec, prefix):
+        os.chdir("src")
+        install_tree(".", prefix)
+        os.chdir("..")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nvdimmsim/package.py]:
```suggestion
        with working_dir('src'):
            install_tree(".", prefix)
```

File: var/spack/repos/builtin/packages/nvdimmsim/package.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+    def build(self, spec, prefix):
+        os.chdir("src")
+        if spec.satisfies("platform=darwin"):
+            make("libnvdsim.dylib")
+        else:
+            make("libnvdsim.so")
+        os.chdir("..")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nvdimmsim/package.py]:
```suggestion
        with working_dir('src'):
            if spec.satisfies("platform=darwin"):
                make("libnvdsim.dylib")
            else:
                make("libnvdsim.so")
```

File: var/spack/repos/builtin/packages/ramulator/package.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+class Ramulator(MakefilePackage):
+    """
+    Ramulator is a fast and cycle-accurate DRAM simulator that supports
+    a wide array of commercial, as well as academic, DRAM standards.
+    """
+
+    homepage = "https://github.com/CMU-SAFARI/ramulator"
+    git = "https://github.com/CMU-SAFARI/ramulator"
+
+    version('sst', git="git@github.com:CMU-SAFARI/ramulator.git",
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ramulator/package.py]: `git` is not needed here since you define it above

File: var/spack/repos/builtin/packages/hybridsim/package.py (hunk 7 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/hybridsim/package.py:30]: I'm not the author of the package. That's just how it's set up... I would happily change the package to work more sensibly if I owned it. It expects you to either symlink or copy those folders into its folder.

File: var/spack/repos/builtin/packages/goblin-hmc-sim/package.py (hunk 6 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/goblin-hmc-sim/package.py]: Touché

File: var/spack/repos/builtin/packages/dramsim2/package.py (hunk 5 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/dramsim2/package.py]: Sorry, don't quite understand. Move it out of the `version` call? Or just move it up? It's the only URL right now, but doesn't it make more sense to leave it where it is if it's version-specific?

File: var/spack/repos/builtin/packages/nvdimmsim/package.py (hunk 9 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/nvdimmsim/package.py]: Yep, that's better.

File: var/spack/repos/builtin/packages/nvdimmsim/package.py (hunk 10 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/nvdimmsim/package.py]: Yep, that's better, thank you.

File: var/spack/repos/builtin/packages/ramulator/package.py (hunk 11 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/ramulator/package.py]: Touché

File: var/spack/repos/builtin/packages/dramsim2/package.py (hunk 5 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/dramsim2/package.py]: All URLs are version-specific, but Spack applies magic to parse the URL, so you only need to list a single URL for multiple versions:
```console
$ spack url parse https://github.com/dramninjasUMD/DRAMSim2/archive/v2.2.2.tar.gz
==> Parsing URL: https://github.com/dramninjasUMD/DRAMSim2/archive/v2.2.2.tar.gz

==> Matched version regex  1: r'^v?(\\d[\\d._-]*)$'
==> Matched  name   regex  0: r'github\\.com/[^/]+/([^/]+)'

==> Detected:
    https://github.com/dramninjasUMD/DRAMSim2/archive/v2.2.2.tar.gz
                                     --------          ~~~~~
    name:    DRAMSim2
    version: 2.2.2

==> Substituting version 9.9.9b:
    https://github.com/dramninjasUMD/DRAMSim2/archive/v9.9.9b.tar.gz
                                     --------          ~~~~~~
```
So yes, move it out of the `version` call and have it be a top-level variable. This allows things like `spack versions` and `spack checksum` to locate new versions.

Comment (jjwilke) [reply var/spack/repos/builtin/packages/dramsim2/package.py]: I made the change. It feels a bit weird to trust that all versions will be located in the same place and with the same file format - but I guess this is rarely an issue and can always be undone with `url=` in the `version` statement (as is done in `intel-pin`).

**Comment by jjwilke** [comment]
@adamjstewart I think this is ready to go? Thanks for your help.

File: var/spack/repos/builtin/packages/hbm-dramsim2/package.py [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    """
+
+    homepage = "https://github.com/tactcomplabs/HBM"
+    git = "https://github.com/tactcomplabs/HBM"
+    url = "https://github.com/tactcomplabs/HBM/archive/hbm-1.0.0-release.tar.gz"
+
+    version('1.0.0', sha256="0efad11c58197edb47ad1359f8f93fb45d882c6bebcf9f2143e0df7a719689a0")
+
+    def build(self, spec, prefix):
+        make()
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hbm-dramsim2/package.py]: No need to override `build` here, this is the default behavior.

File: var/spack/repos/builtin/packages/hybridsim/package.py [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    url = "https://github.com/jimstevens2001/HybridSim/archive/v2.0.1.tar.gz"
+
+    version('2.0.1', sha256="57b82ac929acd36de84525e4d61358f1ab6532f5b635ca3f560e563479921937")
+
+    depends_on("dramsim2")
+    depends_on("nvdimmsim")
+    patch("makefile.patch", when="@2.0.1")
+
+    def install(self, spec, prefix):
+        install_tree(".", prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hybridsim/package.py]: I would move `install` below `build`

File: var/spack/repos/builtin/packages/nvdimmsim/package.py [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+    homepage = "https://github.com/slunk/NVDIMMSim"
+    git = "https://github.com/slunk/NVDIMMSim"
+    url = "https://github.com/jimstevens2001/NVDIMMSim/archive/v2.0.0.tar.gz"
+
+    version('2.0.0', sha256="2a621ef10be5e52a1f543985d08354a2e6ee6532b5720e5f17ad6362cfd4adef")
+
+    def install(self, spec, prefix):
+        with working_dir("src"):
+            install_tree(".", prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nvdimmsim/package.py]: I would move `install` below `build`

File: var/spack/repos/builtin/packages/ramulator/package.py [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    homepage = "https://github.com/CMU-SAFARI/ramulator"
+    git = "https://github.com/CMU-SAFARI/ramulator"
+
+    version('sst', commit="7d2e72306c6079768e11a1867eb67b60cee34a1c")
+
+    patch('ramulator_sha_7d2e723_gcc48Patch.patch', when="@sst")
+    patch('ramulator_sha_7d2e723_libPatch.patch', when="@sst")
+
+    def install(self, spec, prefix):
+        install_tree(".", prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ramulator/package.py]: Same here

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 16]
```diff
@@ -7,38 +7,72 @@
...
 
-    homepage = "http://sst-simulator.org/"
-    url      = "https://github.com/sstsimulator/sst-core/releases/download/v8.0.0_Final/sstcore-8.0.0.tar.gz"
-    git      = "https://github.com/sstsimulator/sst-core.git"
+    homepage = "https://github.com/sstsimulator"
+    git = "https://github.com/sstsimulator/sst-core.git"
+    url = "https://github.com/sstsimulator/sst-core/releases/download/v9.1.0_Final/sstcore-9.1.0.tar.gz"
 
-    version('develop', branch='devel')
-    version('8.0.0', sha256='34a62425c3209cf80b6bca99cb0dcc328b67fb84ed92d5e6d6c975ad9319ba8a')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-core/package.py:19]: We don't normally remove old releases of packages from Spack in case someone needs to reproduce old builds

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 17]
```diff
@@ -7,38 +7,72 @@
...
+    version('9.1.0', sha256="cfeda39bb2ce9f32032480427517df62e852c0b3713797255e3b838075f3614d")
+    version('devel',   branch='devel')
+    version('master',  branch='master', preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-core/package.py]: Is there a reason why 9.1.0 can't be preferred? We like to install checksummable releases by default

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 18]
```diff
@@ -7,38 +7,72 @@
...
 
-    variant('mpi', default=True, description='Support multi-node simulations using MPI')
-    variant('boost', default=False, description='Use boost')
+    variant("pdes-mpi", default=True,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-core/package.py]: I try to avoid variants with dashes in the name, since variants can also be disabled with `-foo` and it might confuse users. Maybe we could name it `pdes`?

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 19]
```diff
@@ -7,38 +7,72 @@
...
+    variant("zlib",    default=False,
+            description="Build support for ZLIB compression")
+    variant("preview", default=False,
+            description="Preview build with deprecated features removed")
 
-    depends_on('autoconf@1.68:', type='build', when='@develop')
-    depends_on('automake@1.11.1:', type='build', when='@develop')
-    depends_on('libtool@1.2.4:', type='build', when='@develop')
-    depends_on('m4', type='build', when='@develop')
+    depends_on("python")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-core/package.py]: This is usually type=build/run

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 20]
```diff
@@ -7,38 +7,72 @@
...
-    url      = "https://github.com/sstsimulator/sst-core/releases/download/v8.0.0_Final/sstcore-8.0.0.tar.gz"
-    git      = "https://github.com/sstsimulator/sst-core.git"
+    homepage = "https://github.com/sstsimulator"
+    git = "https://github.com/sstsimulator/sst-core.git"
+    url = "https://github.com/sstsimulator/sst-core/releases/download/v9.1.0_Final/sstcore-9.1.0.tar.gz"
 
-    version('develop', branch='devel')
-    version('8.0.0', sha256='34a62425c3209cf80b6bca99cb0dcc328b67fb84ed92d5e6d6c975ad9319ba8a')
+    version('9.1.0', sha256="cfeda39bb2ce9f32032480427517df62e852c0b3713797255e3b838075f3614d")
+    version('devel',   branch='devel')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-core/package.py]:
```suggestion
    version('develop',   branch='devel')
```
This allows Spack to sort the version correctly

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 21]
```diff
@@ -7,38 +7,72 @@
...
-    depends_on('mpi', when='+mpi')
-    depends_on('boost@1.56.0:', type='build', when='+boost')
+    depends_on('autoconf@1.68:', type='build', when='@devel')
+    depends_on('automake@1.11.1:', type='build', when='@devel')
+    depends_on('libtool@1.2.4:', type='build', when='@devel')
+    depends_on('m4', type='build', when='@devel')
+    depends_on('autoconf@1.68:', type='build', when='@master')
+    depends_on('automake@1.11.1:', type='build', when='@master')
+    depends_on('libtool@1.2.4:', type='build', when='@master')
+    depends_on('m4', type='build', when='@master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-core/package.py]:
```suggestion
    depends_on('autoconf@1.68:', type='build', when='@master:')
    depends_on('automake@1.11.1:', type='build', when='@master:')
    depends_on('libtool@1.2.4:', type='build', when='@master:')
    depends_on('m4', type='build', when='@master:')
```

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 22]
```diff
@@ -7,38 +7,72 @@
...
 
     def configure_args(self):
         args = []
-        spec = self.spec
+        if "+zoltan" in self.spec:
+            args.append("--with-zoltan=%s" % self.spec["zoltan"].prefix)
+        if "+hdf5" in self.spec:
+            args.append("--with-hdf5=%s" % self.spec["hdf5"].prefix)
+        if "+zlib" in self.spec:
+            args.append("--with-zlib=%s" % self.spec["zlib"].prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-core/package.py:62]: Is it possible to explicitly disable these in an else clause?

File: var/spack/repos/builtin/packages/sst-elements/package.py [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+       architecture simulation from node-level to system-level using
+       the SST discrete event core
+    """
+
+    homepage = "https://github.com/sstsimulator"
+    git = "https://github.com/sstsimulator/sst-elements.git"
+    url = "https://github.com/sstsimulator/sst-elements/releases/download/v9.1.0_Final/sstelements-9.1.0.tar.gz"
+
+    version('9.1.0', sha256="e19b05aa6e59728995fc059840c79e476ba866b67887ccde7eaf52a18a1f52ca")
+    version('devel',   branch='devel')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-elements/package.py]:
```suggestion
    version('develop',   branch='devel')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
If you add:
```python
maintainers = ['jjwilke']
```
to all of these packages, you can be an official "maintainer". This gives us someone to ping in case a user reports build issues or submits a PR to modify these packages.

File: var/spack/repos/builtin/packages/sst-core/package.py (hunk 16 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/sst-core/package.py:19]: We never tested the old versions. The `sst-core` and `sst-macro` packages never had official support. We'd prefer that this PR wipe out whatever was there. I can pretty much guarantee no one ever used what was there.

File: var/spack/repos/builtin/packages/sst-core/package.py (hunk 17 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/sst-core/package.py]: Yes, done. We were mostly working with master, but our 10.0 release will replace 9.1 as preferred soon anyway.

File: var/spack/repos/builtin/packages/sst-core/package.py (hunk 18 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/sst-core/package.py]: It needs to have the mpi in there to distinguish from pdes with threads. I made it an underscore, which appears to be common in a lot of packages.

File: var/spack/repos/builtin/packages/sst-core/package.py (hunk 19 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/sst-core/package.py]: Done.

File: var/spack/repos/builtin/packages/sst-core/package.py (hunk 21 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/sst-core/package.py]: changed for all packages

File: var/spack/repos/builtin/packages/sst-core/package.py (hunk 22 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/sst-core/package.py:62]: It is not. These are implicitly off when the `--with-` is omitted. I agree these might make more sense as a `--enable` and `--disable`, but we are stuck with the build system as is-is.  Our users would not be pleased if we changed a bunch of option names.

File: var/spack/repos/builtin/packages/sst-elements/package.py (hunk 23 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/sst-elements/package.py]: fixed in all packages

File: var/spack/repos/builtin/packages/nvdimmsim/package.py (hunk 14 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/nvdimmsim/package.py]: Fixed in all packages.

File: var/spack/repos/builtin/packages/hybridsim/package.py (hunk 13 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/hybridsim/package.py]: fixed in all packages

File: var/spack/repos/builtin/packages/hbm-dramsim2/package.py (hunk 12 above)
Comment (jjwilke) [reply var/spack/repos/builtin/packages/hbm-dramsim2/package.py]: fixed in all packages

File: var/spack/repos/builtin/packages/ramulator/ramulator_sha_7d2e723_libPatch.patch [hunk 24]
```diff
@@ -0,0 +1,23 @@
+diff --git a/Makefile b/Makefile
+index ea340c8..314113d 100644
+--- a/Makefile
++++ b/Makefile
+@@ -9,7 +9,7 @@ OBJS := $(patsubst $(SRCDIR)/%.cpp, $(OBJDIR)/%.o, $(SRCS))
+ #   g++ 4.x due to an internal compiler error when processing lambda functions.
+ CXX := clang++
+ # CXX := g++-5
+-CXXFLAGS := -O3 -std=c++11 -g -Wall
++CXXFLAGS := -O3 -std=c++11 -g -Wall -fpic
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ramulator/ramulator_sha_7d2e723_libPatch.patch:10]: This will only work for certain compilers, it would be better to query `self.compiler.cxx_pic_flag` and use that. You should have access to `self` in a `patch(self)` method.

File: var/spack/repos/builtin/packages/sst-elements/package.py [hunk 25]
```diff
@@ -0,0 +1,101 @@
...
+    variant("hybridsim", default=False,
+            description="Build with HybridSim support")
+    variant("goblin",    default=False,
+            description="Build with GoblinHMCSim support")
+    variant("hbm",       default=False,
+            description="Build with HBM DRAMSim2 support")
+    variant("ramulator", default=False,
+            description="Build with Ramulator support")
+
+    depends_on("python")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-elements/package.py]: Should this be type=build/run?

**Comment by jjwilke** [comment]
@adamjstewart good to go? thanks again.

File: var/spack/repos/builtin/packages/sst-macro/package.py [hunk 26]
```diff
@@ -16,69 +16,92 @@ class SstMacro(AutotoolsPackage):
...
 
-    version('develop', branch='devel')
-    version('8.0.0', sha256='8618a259e98ede9a1a2ce854edd4930628c7c5a770c3915858fa840556c1861f')
-    version('6.1.0', sha256='930b67313b594148d6356e550ca370214a9283858235321d3ef974191eb028d6')
+    maintainers = ['jjwilke']
 
-    depends_on('boost@1.59:', when='@:6.1.0')
+    version('10.0.0', sha256='064b732256f3bec9b553e00bcbc9a1d82172ec194f2b69c8797f585200b12566')
+    version('master',  branch='master')
+    version('devel', branch='devel')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-macro/package.py]:
```suggestion
    version('develop', branch='devel')
```

File: var/spack/repos/builtin/packages/sst-macro/package.py [hunk 27]
```diff
@@ -16,69 +16,92 @@ class SstMacro(AutotoolsPackage):
...
         spec = self.spec
         args.append(
             '--enable-static=%s' % ('yes' if '+static' in spec else 'no'))
         args.append(
             '--enable-shared=%s' % ('yes' if '+shared' in spec else 'no'))
 
-        if spec.satisfies("@8.0.0:"):
+        if (spec.satisfies("@8.0.0:")
+            or spec.satisfies("@devel")
+                or spec.satisfies("@master")):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-macro/package.py]: Both `develop` and `master` satisfy `@8.0.0:` so you can remove those two from this if-statement.

File: var/spack/repos/builtin/packages/sst-macro/package.py [hunk 28]
```diff
@@ -16,69 +16,92 @@ class SstMacro(AutotoolsPackage):
...
+                '--%sable-multithread' %
+                ('en' if '+pdes_threads' in spec else 'dis')
             ])
 
             if '+skeletonizer' in spec:
                 args.append('--with-clang=' + spec['llvm'].prefix)
 
+        if (spec.satisfies("@10:")
+            or spec.satisfies("@devel")
+                or spec.satisfies("@master")):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-macro/package.py]: Same here

File: var/spack/repos/builtin/packages/sst-transports/package.py [hunk 29]
```diff
@@ -0,0 +1,25 @@
...
+class SstTransports(CMakePackage):
+    """Provides transports like uGNI and verbs
+       that run in the simulator"""
+
+    homepage = "https://github.com/sstsimulator"
+    git = "https://github.com/jjwilke/sst-transports.git"
+
+    maintainers = ['jjwilke']
+
+    version('master',  branch='master', preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sst-transports/package.py]:
```suggestion
    version('master',  branch='master')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]
